04-single-hop-6lowpan-icmp/test_spec04.py

```diff
@@ -0,0 +1,282 @@
+    pinger.ifconfig_set(pinger_netif, "channel", 26)
+
+    res = ping6(pinger, pinged_addr,
+                count=1000, interval=20, packet_size=0)
+    assert res['stats']['packet_loss'] < 10
+
+    assert pktbuf(pinged).is_empty()
+    assert pktbuf(pinger).is_empty()
+
+
+@pytest.mark.iotlab_creds
+# nodes passed to riot_ctrl fixture
+@pytest.mark.parametrize('nodes',
+                         [pytest.param(['samr21-xpro', 'iotlab-m3'])],
+                         indirect=['nodes'])
+def test_task02(riot_ctrl):
+    pinger, pinged = (
+        riot_ctrl(0, APP, Shell, modules=["gnrc_pktbuf_cmd"]),
+        riot_ctrl(1, APP, Shell, modules=["gnrc_pktbuf_cmd"]),
+    )
+
+    pinged_netif, _ = lladdr(pinged.ifconfig_list())
+    pinged.ifconfig_set(pinged_netif, "channel", 17)
+    pinger_netif, _ = lladdr(pinger.ifconfig_list())
+    pinger.ifconfig_set(pinger_netif, "channel", 17)
+
+    res = ping6(pinger, "ff02::1",
+                count=1000, interval=100, packet_size=50)
+    assert res['stats']['packet_loss'] < 10
+
+    assert pktbuf(pinged).is_empty()
+    assert pktbuf(pinger).is_empty()
+
+
+@pytest.mark.iotlab_creds
+# nodes passed to riot_ctrl fixture
+@pytest.mark.parametrize('nodes',
+                         [pytest.param(['iotlab-m3', 'iotlab-m3'])],
+                         indirect=['nodes'])
+def test_task03(riot_ctrl):
+    pinger, pinged = (
+        riot_ctrl(0, APP, Shell, modules=["gnrc_pktbuf_cmd"]),
+        riot_ctrl(1, APP, Shell, modules=["gnrc_pktbuf_cmd"]),
+    )
+
+    pinged_netif, pinged_addr = lladdr(pinged.ifconfig_list())
+    pinged.ifconfig_set(pinged_netif, "channel", 26)
+    assert pinged_addr.startswith("fe80::")
+    pinger_netif, _ = lladdr(pinger.ifconfig_list())
+    pinger.ifconfig_set(pinger_netif, "channel", 26)
+
+    res = ping6(pinger, pinged_addr,
+                count=500, interval=300, packet_size=1024)
+    assert res['stats']['packet_loss'] < 10
+
+    assert pktbuf(pinged).is_empty()
+    assert pktbuf(pinger).is_empty()
+
+
+@pytest.mark.iotlab_creds
+# nodes passed to riot_ctrl fixture
+@pytest.mark.parametrize('nodes',
+                         [pytest.param(['samr21-xpro', 'iotlab-m3'])],
+                         indirect=['nodes'])
+def test_task04(riot_ctrl):
+    pinger, pinged = (
+        riot_ctrl(0, APP, Shell, modules=["gnrc_pktbuf_cmd"]),
+        riot_ctrl(1, APP, Shell, modules=["gnrc_pktbuf_cmd"]),
+    )
+
+    pinged_netif, pinged_addr = lladdr(pinged.ifconfig_list())
+    pinged.ifconfig_set(pinged_netif, "channel", 26)
+    assert pinged_addr.startswith("fe80::")
+    pinger_netif, _ = lladdr(pinger.ifconfig_list())
+    pinger.ifconfig_set(pinger_netif, "channel", 26)
+
+    res = ping6(pinger, pinged_addr,
+                count=10000, interval=100, packet_size=100)
+    assert res['stats']['packet_loss'] < 10
+
+    assert pktbuf(pinged).is_empty()
+    assert pktbuf(pinger).is_empty()
+
+
+@pytest.mark.xfail(reason="Experimental task")
```

Comment:
Realizing this while running the tests, but I'm not sure these should be marked as `xfail`, if they are then we get no information about the failure, I think we should not mark them as such.

```
 RIOT_TERMINAL=socat RIOTBASE=/home/francisco/workspace/RIOT TRIBE_CI=1 pytest 04-single-hop-6lowpan-icmp/test_spec04.py -k="test_task05 or test_task06 or test_task07 or test_task08" --local
========================================================== test session starts ==========================================================
platform linux -- Python 3.6.9, pytest-5.4.3, py-1.9.0, pluggy-0.13.1 -- /home/francisco/.virtualenvs/release/bin/python3
cachedir: .pytest_cache
rootdir: /home/francisco/workspace/Release-Specs, inifile: setup.cfg
collected 10 items / 6 deselected / 4 selected

04-single-hop-6lowpan-icmp/test_spec04.py::test_task05[nodes0] XFAIL                                                              [ 25%]
04-single-hop-6lowpan-icmp/test_spec04.py::test_task06[nodes0] XFAIL                                                              [ 50%]
04-single-hop-6lowpan-icmp/test_spec04.py::test_task07[nodes0] XPASS                                                              [ 75%]
04-single-hop-6lowpan-icmp/test_spec04.py::test_task08[nodes0] XFAIL                                                              [100%]

------------------------------ generated xml file: /home/francisco/workspace/Release-Specs/test-report.xml ------------------------------
======================================== 6 deselected, 3 xfailed, 1 xpassed in 292.06s (0:04:52) ========================================
```

For now what I would rather do is mark `05-08` as local only since `arduino-zero` is currently broken in iotlab, that way they will be skipped anyway

Reply:
Will do. There are however also other tests in other specs that are marked as `xfail` due to their experimental nature. Shall I remove them there as well?

Reply:
I think we should, can you point me to the others that were marked?

Reply:
Appearently the only one remaining is in https://github.com/RIOT-OS/Release-Specs/pull/160:

```
03-single-hop-ipv6-icmp/test_spec03.py:@pytest.mark.xfail(reason="See https://github.com/RIOT-OS/RIOT/issues/12565")
```